model/task_queue.go

```diff
@@ -155,34 +155,44 @@ func ValidateNewGraph(t *task.Task, tasksToBlock []string) error {
 
 func (self *TaskQueue) BlockTaskGroupTasks(spec TaskSpec, taskID string) error {
 	catcher := grip.NewBasicCatcher()
-	tasksToBlock := []string{}
-	for _, it := range self.Queue {
-		if spec.Group == it.Group && spec.BuildVariant == it.BuildVariant && spec.Version == it.Version {
-			tasksToBlock = append(tasksToBlock, it.Id)
-		}
-	}
+	tasksToBlock := []task.Task{}
 	t, err := task.FindOneId(taskID)
 	if err != nil {
 		return errors.Wrapf(err, "problem finding task %s", taskID)
 	}
 	if t == nil {
 		return errors.Errorf("found nil task %s", taskID)
 	}
+
+	p, err := FindProjectFromTask(t)
+	if err != nil {
+		return errors.Wrapf(err, "problem getting project for task %s", t.Id)
+	}
+	tg := p.FindTaskGroup(t.TaskGroup)
+	encounteredTask := false
+	for _, tgTask := range tg.Tasks {
```

Comment:
is it possible to express this logic as an aggregation operation...

Reply:
This logic is fairly complex. I'm saying, given an array like `["a", "b", "c", "d", "e"]` and a string like `"c"`, construct a query from all array elements with indexes greater than `"c"`'s.